src/App.tsx

```diff
@@ -17,18 +79,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onSelectStatus={selectStatus}
+                status={status}
+                onFilter={setFilterQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length
+                ? (
+                  <TodoList
+                    todos={visibleTodos}
+                    onSelectTodoId={selectTodoId}
+                    selectedTodoId={selectedTodoId}
+                  />
+                )
+                : <Loader />}
```

Comment:
But what if the download was successful but there are simply no todos on the server?